source/classes.tex

```diff
@@ -6642,28 +6642,54 @@
 declared in the \grammarterm{member-specification} of \tcode{C}
 that is
 \begin{itemize}
-\item a non-static member of \tcode{C} having one parameter of type \tcode{const C\&}, or
+\item a non-static const member of \tcode{C} having one parameter of type \tcode{const C\&}, or
 \item a friend of \tcode{C} having two parameters of type \tcode{const C\&}.
 \end{itemize}
 
 \pnum
-\indextext{structural comparison operator|see{operator, structural comparison}}%
-A three-way comparison operator for a class type \tcode{C}
-is a \defnadj{structural comparison}{operator}
-if it is defined as defaulted in the definition of \tcode{C},
-and all three-way comparison operators it invokes
-are structural comparison operators.
-\indextext{strong structural equality|see{equality, strong structural}}%
-A type \tcode{T}
-has \defnadj{strong structural}{equality}
-if, for a glvalue \tcode{x} of type \tcode{const T},
-\tcode{x <=> x} is a valid expression
-of type \tcode{std::strong_ordering} or \tcode{std::strong_equality}
-and either does not invoke a three-way comparison operator
-or invokes a structural comparison operator.
+If the class definition
+does not explicitly declare an \tcode{==} operator function,
+but declares a defaulted three-way comparison operator function,
+an \tcode{==} operator function is declared implicitly
+with the same access as the three-way comparison operator function.
+The implicitly-declared \tcode{==} operator for a class \tcode{X}
+is an inline member and is defined as defaulted in the definition of \tcode{X}.
+If the three-way comparison operator function
+is declared as a non-static const member,
+the implicitly-declared \tcode{==} operator function is a member of the form
+\begin{codeblock}
+bool X::operator==(const X&) const;
+\end{codeblock}
+Otherwise, the implicitly-declared \tcode{==} operator function is of the form
+\begin{codeblock}
+friend bool operator==(const X&, const X&);
+\end{codeblock}
+\begin{note}
+Such a friend function is not visible to unqualified lookup\iref{namespace.memdef}.
```

Comment:
We should leave the cross-reference to [namespace.memdef] in, because that's curiously the only place where we clearly say how friend name injection works.
Otherwise addressed, including moving "only" to the end.